packages/components/pie-form-label/test/component/pie-form-label.spec.ts

```diff
@@ -24,4 +26,77 @@ test.describe('PieFormLabel - Component tests', () => {
     });
+    test.describe('should focus the component assigned with "for" when clicked', () => {
+        // IMPORTANT: Mounting and Unmounting the component before each test ensures that any tests that do not explicitly
+        // mount the component will still have it available in Playwright's cache (loaded and registered in the test browser)
+        test.beforeEach(async ({ mount }) => {
+            const pieInput = await mount(PieInput);
+            await pieInput.unmount();
+
+            const pieSwitch = await mount(PieSwitch);
+            await pieSwitch.unmount();
+
+            const pieFormLabel = await mount(PieFormLabel);
+            await pieFormLabel.unmount();
+        });
+
+        test('when used with a text input', async ({ page }) => {
+            // Arrange
+            await page.setContent('<pie-form-label for="email">Email:</pie-form-label><pie-input id="email"></pie-input>');
+
+            // Act
+            const target = page.locator('#email');
+            const label = page.locator('pie-form-label');
+            await label.click();
+
+            // Assert
+            expect(target).toBeFocused();
+        });
+
+        test.describe('when used with a switch', () => {
+            const markup = '<pie-form-label for="approveSettings">Approve settings</pie-form-label><pie-switch id="approveSettings"></pie-switch>';
+
+            test.describe('when clicked once', () => {
+                test('the switch should be focused', async ({ page }) => {
+                    // Arrange
+                    await page.setContent(markup);
+
+                    // Act
+                    const target = page.locator('#approveSettings');
+                    const label = page.locator('pie-form-label');
+                    await label.click();
+
+                    // Assert
+                    expect(target).toBeFocused();
+                });
+                test('the switch "checked" attribute is true', async ({ page }) => {
+                    // Arrange
+                    await page.setContent(markup);
+
+                    // Act
+                    const target = page.locator('#approveSettings');
+                    const label = page.locator('pie-form-label');
+                    await label.click();
+
+                    // Assert
+                    expect(target).toBeFocused();
+                });
+            });
+            test.describe('when clicked twice', () => {
+                test('the switch "checked" attribute is false', async ({ page }) => {
+                    // Arrange
+                    await page.setContent(markup);
+
+                    // Act
+                    const target = page.locator('#approveSettings');
+                    const label = page.locator('pie-form-label');
+                    await label.click();
+                    await label.click();
+
+                    // Assert
+                    expect(target).toBeFocused();
```

Comment:
I just noticed that this test is checking the focus state instead of the checked state, just in case you also missed it

Reply:
I fixed that one too. Thanks! 🙇‍♂️